### ISSUE TITLE
Libreoffice: Errors when navigating through the document, NVDA does not report text

### ISSUE THREAD

**Issue by Adriani90** [nvaccess/nvda#11191]
### Steps to reproduce:
1. Open NVDA and open Libreoffice writer
2. Type "hello" and press enter two times
3. Press ctrl+enter to add a page break
4. Press enter two times
5. Press up arrow or left arrow multiple times very fast

### Actual behavior:
NVDA generates lots of errors and does not report text.
Following error is in the log:

```
IO - inputCore.InputManager.executeGesture (16:58:34.276) - winInputHook (812):
Input: kb(laptop):upArrow
ERROR - eventHandler.executeEvent (16:58:34.332) - MainThread (13060):
error executing event: caret on <NVDAObjects.Dynamic_SymphonyParagraphSymphonyTextEditableTextWithAutoSelectDetectionIAccessible object at 0x01567F50> with extra args of {}
Traceback (most recent call last):
  File "eventHandler.pyc", line 213, in executeEvent
  File "eventHandler.pyc", line 94, in __init__
  File "eventHandler.pyc", line 102, in next
  File "compoundDocuments.pyc", line 469, in event_caret
  File "documentBase.pyc", line 24, in makeTextInfo
  File "compoundDocuments.pyc", line 216, in __init__
  File "documentBase.pyc", line 24, in makeTextInfo
  File "textInfos\offsets.pyc", line 453, in __init__
  File "NVDAObjects\IAccessible\__init__.pyc", line 187, in _getCaretOffset
RuntimeError: no active caret in this object
```

### Expected behavior:
No error and NVDA should report text when navigating with arrow keys.

#### NVDA installed/portable/running from source:
Installed

#### NVDA version:
2020.1

#### Windows version:
Windows 10 1909 Update

#### Name and version of other software in use when reproducing the issue:
Libreoffice writer 6.4.4.2

#### Does the issue still occur after restarting your computer?
yes

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
no

#### If addons are disabled, is your problem still occuring?
yes

#### Did you try to run the COM registry fixing tool in NVDA menu / tools?
yes, no effect

**Comment by DrSooom** [comment]
I cannot reproduce it with NVDA 2019.2.1 (installed), NVDA 2019.3.1 (portable) and NVDA Alpha-20220 (portable) on Win7x64. NVDA reads "blank" ("Leer" in German) and the letters so far correctly, except the fact that the page break is also spoken as "blank".

I only got the following error in the NVDA Alpha-20220 log:
```
IO - inputCore.InputManager.executeGesture (19:51:14.105) - winInputHook (10060):
Input: kb(desktop):leftArrow
DEBUGWARNING - NVDAObjects.IAccessible.IA2TextTextInfo._getCaretOffset (19:51:14.162) - MainThread (1808):
IAccessibleText::caretOffset failed
Traceback (most recent call last):
  File "NVDAObjects\IAccessible\__init__.pyc", line 182, in _getCaretOffset
  File "comtypesMonkeyPatches.pyc", line 26, in __call__
_ctypes.COMError: (-2147467259, 'Unbekannter Fehler', (None, None, None, 0, None))
DEBUG - editableText.EditableText._hasCaretMoved (19:51:14.181) - MainThread (1808):
Focus event. Elapsed: 10 ms
IO - speech.speak (19:51:14.191) - MainThread (1808):
Speaking ['Leer']
```

**Comment by Adriani90** [comment]
Which version of LibreOffice did you test with?

 

 

Von: Daniel Mayr <notifications@github.com> 
Gesendet: Sonntag, 24. Mai 2020 20:05
An: nvaccess/nvda <nvda@noreply.github.com>
Cc: Adriani90 <adriani.botez@googlemail.com>; Author <author@noreply.github.com>
Betreff: Re: [nvaccess/nvda] Libreoffice: Errors when navigating through the document, NVDA does not report text (#11191)

 

I cannot reproduce it with NVDA 2019.2.1 (installed), NVDA 2019.3.1 (portable) and NVDA Alpha-20220 (portable) on Win7x64. NVDA reads "blank" ("Leer" in German) and the letters so far correctly, except the fact that the page break is also spoken as "blank".

I only got the following error in the NVDA Alpha-20220 log:

IO - inputCore.InputManager.executeGesture (19:51:14.105) - winInputHook (10060):
Input: kb(desktop):leftArrow
DEBUGWARNING - NVDAObjects.IAccessible.IA2TextTextInfo._getCaretOffset (19:51:14.162) - MainThread (1808):
IAccessibleText::caretOffset failed
Traceback (most recent call last):
  File "NVDAObjects\IAccessible\__init__.pyc", line 182, in _getCaretOffset
  File "comtypesMonkeyPatches.pyc", line 26, in __call__
_ctypes.COMError: (-2147467259, 'Unbekannter Fehler', (None, None, None, 0, None))
DEBUG - editableText.EditableText._hasCaretMoved (19:51:14.181) - MainThread (1808):
Focus event. Elapsed: 10 ms
IO - speech.speak (19:51:14.191) - MainThread (1808):
Speaking ['Leer']

**Comment by DrSooom** [comment]
LibreOffice 6.4.4.2 is installed.

Check in the LibreOffice options in the tab "Accessibility" if the support for AT software is enabled. Then uncheck the entries "Sidebar" (CTRL+F5, in German "Seitenleiste") and "Navigator" in the "View" menu in LibreOffice Writer. And then maximize the window and set the zoom level (in German "Maßstab") to "Page width".

**Comment by Adriani90** [comment]
@DrSooom are you saying that without these settings you can reproduce this issue?
Note that an user should not be forced to do these settings to avoid errors.

**Comment by Adriani90** [comment]
cc: @lukaszgo1

**Comment by DrSooom** [comment]
>@DrSooom are you saying that without these settings you can reproduce this issue?

No, I didn't.

The very first option I mentioned should be enabled by default if LibreOffice is able to detect NVDA on startup of LibreOffice – if I'm not wrong. So please follow my additional suggestions and test this on your own machine – to limit this issue. Thanks.

**Comment by Adriani90** [comment]
@DrSooom I tried your suggestions and the error still persists. Note this is only reproducible when you press up and down arrow very fast. It seems it is related to textinfo, you can also reproduce a similar error if you press f6 twice and land on settings window. Then press the arrow keys.

cc: @leonardder

**Comment by DrSooom** [comment]
If you have the time it would be great to know if this issue still exists in [LibreOffice 7.0.0 Beta 2](https://www.libreoffice.org/download/download/?version=7.0.0) ([Release Notes](https://wiki.documentfoundation.org/Releases/7.0.0/Beta2)). Otherwise please upload a logfile at Debug level, so that we exactly know what you mean with "very fast". Thanks.

**Comment by Adriani90** [comment]
Thanks Daniel for the link. I tested and it seems now NVDA regains the carret position faster now, it reads the text. But the error is still occuring and I guess this might have performance impact, especially in big documents. It would be nice if someone can investigate.

Just press and hold down ctrl+down arrow or only down arrow as described in the description and you will reproduce the error.

**Comment by DrSooom** [comment]
> 5. Press up arrow or left arrow multiple times very fast

> Just press and hold down ctrl+down arrow or only down arrow as described in the description and you will reproduce the error.

"holding down" and "pressing very fast" are to different things. Therefore I asked for a logfile. When I'm holding down here ArrowRight, when the cursor was located at the beginning of the document before, sometimes "Unknown" instead of the document content is shown on the braille display. Holding down an arrow key while NVDA is running is often not a good idea – depending on the used TTS synthesizer and braille display.

**Comment by Adriani90** [comment]
anyway, the error should not appear.
 no

Von meinem iPhone gesendet

**Comment by Qchristensen** [comment]
I can't replicate this with NVDA 2021.1 RC 1 on LibreOffice Writer 7.0.4.2 or 7.1.4.2 (64-bit) running on Windows 10 (64-bit) Version: 2009, Build: 19043.

@Adriani90, or anyone, are you still able to reproduce, or can we close this as WorksForMe?

**Comment by Adriani90** [comment]
Yeah this seems fixed, I cannot reproduce in LO dev 25.2 with NVDA 2024.3 Beta. Closing as works for me.
